tools/analyzer_plugin/test/integration/diagnostics/non_null_safe/missing_required_prop_test.dart

```diff
@@ -138,16 +140,26 @@ class MissingRequiredPropTest_NoErrors extends MissingRequiredPropTest {
         test() => InheritsLateRequired()();
```

Comment:
This existing test case and the newly-added regression test, together, validate the behavior we want to see.